### ISSUE TITLE
Test with Inf but train normally

### ISSUE THREAD

**Issue by liaocs2008** [HolmesShuan/ResNet-18-Caffemodel-on-ImageNet#15]
Hi,

I tested your model and got following results.

> I1204 15:02:51.487152 20681 caffe.cpp:309] Loss: 1.29169
I1204 15:02:51.487195 20681 caffe.cpp:321] acc/top-1 = 0.68608
I1204 15:02:51.487213 20681 caffe.cpp:321] acc/top-5 = 0.887702
I1204 15:02:51.487231 20681 caffe.cpp:321] loss = 1.29169 (* 1 = 1.29169 loss)

So far so good. Then, I was trying to apply some pruning technique to your model. 

The training loss converges from 7.32351 to 1.74956, but the test loss remains 87.2697 since the beginning. As expected, the test accuracy is always ~0.005 for top-5 and ~0.001 for top-1. 

I compared the L2-norm on single batch of training data and single testing data:

> **Test** 
data 186565.44
label 3736.5166
label_data_1_split_0 3736.5166
label_data_1_split_1 3736.5166
label_data_1_split_2 3736.5166
conv1 21728764.0
pool1 13620027.0
pool1_pool1_0_split_0 13620027.0
pool1_pool1_0_split_1 13620027.0
...
fc1000 inf
fc1000_fc1000_0_split_0 inf
fc1000_fc1000_0_split_1 inf
fc1000_fc1000_0_split_2 inf
loss 87.33654
acc/top-1 0.0
acc/top-5 0.0

> **Train**
data 299836.56
label 6460.825
conv1 2865.9963
pool1 1788.9178
pool1_pool1_0_split_0 1788.9178
pool1_pool1_0_split_1 1788.9178
res2a_branch1 1357.2897
res2a_branch2a 822.31744
res2a_branch2b 1204.2449
...
fc1000 783.6228
loss 1.5208726

Do you have any idea what is the problem? Any feedback would be appreciated. Thanks!

**Comment by HolmesShuan** [comment]
As far as I can tell, if the training process runs normally and testing performs poorly, you may set
```prototxt
batch_norm_param {
      moving_average_fraction: 0.9 
      use_global_stats=false
}
```
for all BatchNorm layers to verify that it is the problem of BN layers.  If `Inf` still exists, please make sure that you are using the same parameters for training and testing, since pruning usually requires the weight pruning step before testing.

**Comment by liaocs2008** [comment]
It's the problem of BN layer. 

Thanks a lot!
